packages/frontend/src/hooks/useLoginWithServer.ts

```diff
@@ -1,27 +1,35 @@
 import { UserState } from '@unirep/core'
+import ERROR_MESSAGES from '../constants/error-messages/loginErrorMessage'
 
 const useLoginWithServer = (
-    navigate: (path: string) => void,
+    accessToken: string | null,
     hashUserId: string | null,
     signMsg: string | null,
+    navigate: (path: string) => void,
+    setErrorCode: (errorCode: keyof typeof ERROR_MESSAGES) => void,
+    setIsLogin: (param: any) => void,
     createUserState: () => Promise<UserState | undefined>
 ) => {
     const loginWithServer = async () => {
         try {
             if (!hashUserId) {
-                throw new Error('No hash user id')
+                throw new Error('MISSING_ELEMENT')
```

Comment:
I thought the error should be chosen among `ERROR_MESSAGES`
e.g. `ERROR_MESSAGES.MISSING_ELEMENT.code`
or in the future, these strings will easily cause typo
and other developers cannot find the error codes easily, they need to refer to other file or document

Reply:
Just changed here 
https://github.com/social-tw/social-tw-website/pull/115/commits/2a67e7cb6736d0f22ec7b96c4200e4e269ae1336